frontend/src/components/SearchBar.tsx

```diff
@@ -0,0 +1,63 @@
+import React, { useState, useEffect } from 'react';
+import { Link } from 'react-router-dom';
+import { useApolloClient } from '@apollo/client';
+import gql from 'graphql-tag';
+
+const SEARCH_QUERY = gql`
+  query SearchByClassTitle($title: String!) {
+    SearchByClassTitle(term: $title) {
+      id
+      classTitle
+      imagePath
+    }
+  }
+`;
+
+const SearchBar = ({ onResults, setSearchTerm, searchTerm }) => {
+  const client = useApolloClient();
+
+  const handleSearch = async () => {
+    try {
+      const { data } = await client.query({
```

Comment:
Never do queries this way in React, use the react query hooks useQuery, useMutation and useSubscription for this.